src/renderers/dom/fiber/__tests__/ReactDOMFiber-test.js

```diff
@@ -43,6 +43,26 @@ describe('ReactDOMFiber', () => {
     expect(container.textContent).toEqual('10');
   });
 
+  it('should be called a callback argument', () => {
```

Comment:
I noticed the stack reconciler also passed the public instance as `this` into the callback.
I found it weird but I guess we should do it here as well for feature parity.

See the [test verifying it](https://github.com/facebook/react/blob/master/src/renderers/dom/stack/client/__tests__/ReactMount-test.js#L260-L298) (it will fail due to missing `unstable_batchedUpdates` but you can temporarily comment it to check if a future fix works).